saleor/payment/migrations/0039_transactionevent_currency.py

```diff
@@ -1,25 +1,25 @@
 from django.db import migrations
-from django.apps import apps as registry
-from django.db.models.signals import post_migrate
+from django.db.models import F, OuterRef, Subquery, QuerySet
 
-from .tasks.saleor3_12 import (
-    set_default_currency_for_transaction_event,
-    set_default_currency_for_transaction_event_task,
-)
-from ... import __version__
+
+def set_default_currency_for_transaction_event(qs: QuerySet):
+    qs.update(currency=F("transaction_currency"))
 
 
 def set_default_currency_for_transaction_event_migration(apps, _schema_editor):
-    def on_migrations_complete(sender=None, **kwargs):
-        set_default_currency_for_transaction_event_task.delay()
-
-    if __version__.startswith("3.12"):
-        sender = registry.get_app_config("payment")
-        post_migrate.connect(on_migrations_complete, weak=False, sender=sender)
-    else:
-        TransactionItem = apps.get_model("payment", "TransactionItem")
-        TransactionEvent = apps.get_model("payment", "TransactionEvent")
-        set_default_currency_for_transaction_event(TransactionItem, TransactionEvent)
+    TransactionItem = apps.get_model("payment", "TransactionItem")
+    TransactionEvent = apps.get_model("payment", "TransactionEvent")
+
+    transaction_item = TransactionItem.objects.filter(
+        pk=OuterRef("transaction")
```

Comment:
Shouldn't be `pk=OuterRef("transaction_id")`?

Reply:
It's equivalent.